apps/vault-apy/src/APYSimulator.ts

```diff
@@ -80,13 +80,55 @@ export default class APYSimulator {
       return;
     }
     const startingTimestamp = startingDate.getTime() / 1000;
+    
     for (let i = 1; i <= numOfDays; i++) {
-      log(`processing day ${i}, ${startingTimestamp - i * ONE_DAY_IN_SECONDS}`);
-      const forkBlock = await this.#getBlockAtTimestamp(
-        startingTimestamp - i * ONE_DAY_IN_SECONDS
-      );
-
-      await this.run(forkBlock);
+      const currentTimestamp = startingTimestamp - i * ONE_DAY_IN_SECONDS;
+      log(`processing day ${i}, ${currentTimestamp}`);
+      const forkBlock = await this.#getBlockAtTimestamp(currentTimestamp);
```

Comment:
It looks like most of this logic is about adjusting the fork block before calling `this.run`. I think this code would be cleaner if you created a function called `getForkBlock` and then put all your adjustments in there. Then the code inside the loop would look like:

```
for (const vault of ... ) {
   const forkBlock = getForkBlock(vault, currentTimestamp)
   await this.run(forkBlock, vault.address)
}
```